x-pack/plugins/canvas/public/services/kibana/data_views.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { KibanaPluginServiceFactory } from '../../../../../../src/plugins/presentation_util/public';
+import { ErrorStrings } from '../../../i18n';
+import { CanvasStartDeps } from '../../plugin';
+import { CanvasDataViewsService } from '../data_views';
+import { CanvasNotifyService } from '../notify';
+
+const { esService: strings } = ErrorStrings;
+
+export type DataViewsServiceFactory = KibanaPluginServiceFactory<
+  CanvasDataViewsService,
+  CanvasStartDeps,
+  {
+    notify: CanvasNotifyService;
+  }
+>;
+
+export const dataViewsServiceFactory: DataViewsServiceFactory = ({ startPlugins }, { notify }) => ({
+  getDataViews: async () => {
+    try {
+      const dataViews = await startPlugins.data.dataViews.getIdsWithTitle();
+      return dataViews.map((view) => view.title);
+    } catch (e) {
+      notify.error(e, { title: strings.getIndicesFetchErrorMessage() });
+    }
+
+    return [];
+  },
+  getFields: async (dataViewTitle: string) => {
```

Comment:
Is this used exclusively when an existing data view doesn't exist? Why are all fields starting with `_` being filtered out? It is valid to have an indexed field that starts with `_`.

Reply:
This is used when we have a title, and need to get the fields.  The `_` fields are filtered out as an attempt to remove meta fields it appears.  Is there something formalized with the DataView api's to filter out meta fields?  Is the `create` the correct route to go when we have only the title and need to fields?

Reply:
Is the user thinking in terms of data views or index pattern strings? If they're using data views then you should get the field list from the data view directly. Lets establish this and then discuss handling meta fields.